### DISCUSSION TITLE
Frequency Adaptive GCN (FAGCN).

### DISCUSSION THREAD

**Discussion by wsad1** [pyg-team/pytorch_geometric#2225] [Ideas]
## 🚀 Feature
FAGCN was proposed in this [paper](https://arxiv.org/abs/2101.00797). 

## Motivation
According to the paper FAGCN works especially well on graphs with low homophily(disassortative). This is due to the fact that it can adaptively use low and high frequency signals in the graph. 
![image](https://user-images.githubusercontent.com/13963626/110419015-8943de00-80be-11eb-8e93-679e5990e8c3.png)

## Additional context
The implementation looks straightforward. But i wanted to get peoples thoughts before creating a PR.

**Comment by rusty1s** [comment]
Very much appreciate a PR on this one :) Let me know if you need any help.

**Reply by wsad1** [reply]
Thanks @rusty1s .  So i'd like some thoughts on the implementation.
Specifically. This is how the full FAGCN layer is implemented. 
![image](https://user-images.githubusercontent.com/13963626/110946521-f7113380-8364-11eb-88a4-709b04aed702.png)
The raw features are transformed , using an MLP, and added to the aggregation at every layer. Now forward could be implemented in 2 ways. 

1. `forward(x,h0)`. Here a user would have to pass node embedding from the previous layer and the **transformed** raw features. And forward would perform all the operations shown in  the image, except for the MLP part.
2. `forward(x)`. Here forward would only perform the aggregation operation and it would be left to the user to implement the rest of it, they would have to add transformed raw features to forwards output, then they would have to do the last Linear transform too.

Which do you think is better.

**Reply by rusty1s** [reply]
I would go with the first option, similar to what we do in [`GCN2Conv`](https://github.com/rusty1s/pytorch_geometric/blob/master/torch_geometric/nn/conv/gcn2_conv.py#L104).

**Reply by wsad1** [reply]
Great, thanks.
